subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/AbstractCompleteFileSystemLocationSnapshot.java

```diff
@@ -41,6 +43,11 @@ public String getName() {
         return name;
     }
 
+    @Override
+    public boolean isSymlink() {
+        return isSymlink;
+    }
```

Comment:
IIUC, this should return `true` for anything that is _accessed via_ a symlink, no? So `isSymlink()` doesn't sound like a good name for it.

Reply:
I am using an enum in https://github.com/gradle/gradle/issues/13053 called `AccessType`, with the values `VIA_SYMLINK` and `DIRECT`. This makes the code clearer.